frontend/src/resources/rbac.ts

```diff
@@ -84,6 +111,47 @@ export interface ClusterRoleBinding {
   }
 }
 
+export interface Role {
+  apiVersion: RbacApiVersionType
+  kind: RoleKindType
+  metadata: Metadata
+  rules: {
+    verbs: string[]
+    apiGroups: string[]
+    resources: string[]
+    resourceNames: string[]
+  }[]
+}
+
+export interface RoleBinding {
+  apiVersion: RbacApiVersionType
+  kind: RoleBindingKindType
+  metadata: Metadata
+  subjects?: {
+    kind: 'User' | 'Group'
+    apiGroup: 'rbac.authorization.k8s.io'
+    name: string
+    namespace: string
+  }[]
+  roleRef: {
+    apiGroup: 'rbac.authorization.k8s.io'
+    kind: RoleKindType
+    name: string
+  }
+}
+
+export interface ServiceAccount {
+  apiVersion: ServiceAccountApiVersionType
+  kind: ServiceAccountKindType
+  metadata: Metadata
+  secrets: string[]
+  imagePullSecrets: string[]
```

Comment:
See https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/service-account-v1/#ServiceAccount so it is something like
```
"imagePullSecrets": [
                {
                    "name": "builder-dockercfg-ggmbc"
                }
            ]
```

so I guess it is fair to explicitly type like
```suggestion
  imagePullSecrets: LocalObjectReference[]
```

and to create a new `kubernetes-client.ts` at `frontend/src/resources` containing

```
export interface LocalObjectReference {
  name: string
}
```

wdyt?